tractor/_actor.py

```diff
@@ -225,11 +254,41 @@ def load_modules(self) -> None:
         the original nursery we need to try and load the local module
         code (if it exists).
         """
-        for path in self.rpc_module_paths:
-            log.debug(f"Attempting to import {path}")
-            self._mods[path] = importlib.import_module(path)
+        try:
+            if self._spawn_method == 'trio_run_in_process':
+                parent_data = self._parent_main_data
+                if 'init_main_from_name' in parent_data:
+                    _mp_fixup_main._fixup_main_from_name(
+                        parent_data['init_main_from_name'])
+                elif 'init_main_from_path' in parent_data:
+                    _mp_fixup_main._fixup_main_from_path(
+                        parent_data['init_main_from_path'])
+
+            for modpath, filepath in self.rpc_module_paths.items():
+                # XXX append the allowed module to the python path which
+                # should allow for relative (at least downward) imports.
+                sys.path.append(os.path.dirname(filepath))
+                # XXX leaving this in for now incase we decide to swap
```

Comment:
Maybe this isn't worth keeping in for now.
I can always dig up how to do it again if someone complains about the path mutating approach.